lib/controller.js

```diff
@@ -641,6 +663,213 @@ controller.tesselFirmwareVerion = function(opts) {
         });
     });
 };
+controller.menu = {
+
+  prepareMenu: function(opts, tessels, resolve, reject, title, callback) {
+    //return new Promise(function(resolve, reject) {
+
+    var rtm = new Menu({
+      //  width: process.stdout.columns - 4,
+      width: 55,
+      x: 1,
+      y: 2,
+      bg: 'red'
+    });
+    rtm.reset();
+    rtm.write(title + '\n');
+    rtm.write('                                                \n');
+
+    // create menu entries
+    for (var i in tessels) {
+      controller.menu.makeMenu(tessels[i], i, rtm);
+    }
+
+    rtm.add('EXIT\n');
+    rtm._draw();
+    if (!opts.menu) {
+      // if this is no test, starting the menu as child process
+      controller.menu.showMenu(rtm, resolve, reject);
+    } else {
+      // because the test needs to resolve the promise ...
+      resolve({
+        opts: opts,
+        tessels: tessels
+      });
+    }
+
+    rtm.once('select', function(label, index) {
+      rtm.close();
+      controller.menu.clear();
+
+      // Identify the Exit command by first letter (all other entries start with numbers/index)
+      if (label[0] !== 'E') {
+        callback(tessels[index].name, index, resolve, reject);
+        // controller.ssh.runSSH(index, opts, tessels, resolve, reject);
+      } else {
+        // going to clear screen and calling exit to resolve promise
+        controller.menu.clear();
+        controller.menu.exit(resolve);
+      }
+    });
+    //  });
+  },
+  makeMenu: function(tessel, index, rtm) {
+    if (tessel.lanConnection) {
+      if (!tessel.lanConnection.authorized) {
+        rtm.add(index + ') ' + tessel.name + ': ' + tessel.lanConnection.ip + ' (not authorized) \n');
+      } else {
+        rtm.add(index + ') ' + tessel.name + ': ' + tessel.lanConnection.ip + ' (authorized) \n');
+      }
+
+    } else if (tessel.usbConnection) {
+      if (!tessel.usbConnection.authorized) {
+        rtm.add(index + ') ' + tessel.name + ': [USB] (not authorized) \n');
```

Comment:
@johnnyman727 OK, thanks, now I understand the security idea. A single Tessel is bond to a single computer after provisioning. I hear ringing the :bell: and see a high frequented FAQ entry:
# Tessel 2 FAQ
> ### My Tessel doesn't work on my second Computer
> For reasons of security your Tessel is only able to `run` and `push` your scripts from a single computer - that one you use `t2 provision`. No other computers are able to connect to your Tessel. But you are able to connect also by LAN from that single computer so you are able to `push` scripts through the air :ghost: :v: 
*-You are visitor 1999*

:see_no_evil: :hear_no_evil: :speak_no_evil: :wink: only my two cents ...

But this is not the worst idea :+1: 

Maybe it would be an idea to provide a solution about using a key-server what accepts sURL based authorization for putting generated keys and use them remotely.

---
Btw: I didn't got this until now, because .... I ... didn't ... own ... a single Tessel 2 ...